src/main/kotlin/org/rust/ide/actions/smartenter/SemicolonFixer.kt

```diff
@@ -0,0 +1,33 @@
+package org.rust.ide.actions.smartenter
+
+import com.intellij.lang.SmartEnterProcessorWithFixers
+import com.intellij.openapi.editor.Editor
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiWhiteSpace
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.lang.core.psi.RustCallExpr
+
+/**
+ * Fixer that adds missing semicolons at the end of statements.
+ */
+class SemicolonFixer : SmartEnterProcessorWithFixers.Fixer<RustSmartEnterProcessor>() {
+    override fun apply(editor: Editor, processor: RustSmartEnterProcessor, element: PsiElement) {
+        if (isOutermostCallExpr(element) && !endsWithSemicolon(element)) {
+            editor.document.insertString(element.textRange.endOffset, ";");
+        }
+    }
+
+    private fun isOutermostCallExpr(element: PsiElement): Boolean {
+        return element is RustCallExpr && PsiTreeUtil.getParentOfType(element, RustCallExpr::class.java) == null;
```

Comment:
We use Kotlin's [Reified type parameters](https://kotlinlang.org/docs/reference/inline-functions.html#reified-type-parameters) to provide a more Kotlinish API for PsiTreeUtil [here](https://github.com/intellij-rust/intellij-rust/blob/master/src/main/kotlin/org/rust/lang/core/psi/util/RustPsiExtensions.kt#L19).

You can write just `element.getParentOfType<RustCallExpr>`.